nucypher/blockchain/eth/sol/source/contracts/StakingEscrow.sol

```diff
@@ -1143,37 +1148,46 @@ contract StakingEscrow is Issuer, IERC900History {
         if (info.nextCommittedPeriod <= previousPeriod && info.nextCommittedPeriod != 0) {
             info.lastCommittedPeriod = info.nextCommittedPeriod;
         }
-        mint(msg.sender);
+        (uint16 processedPeriod1, uint16 processedPeriod2) = mint(msg.sender);
+
+        if (processedPeriod1 != 0 || processedPeriod2 != 0) {
+            policyManager.ping(msg.sender, processedPeriod1, processedPeriod2, 0);
+        }
     }
 
     /**
     * @notice Mint tokens for previous periods if staker locked their tokens and made a commitment
     * @param _staker Staker
+    * @return processedPeriod1 Processed period: currentCommittedPeriod or zero
+    * @return processedPeriod2 Processed period: nextCommittedPeriod or zero
     */
-    function mint(address _staker) internal {
+    function mint(address _staker) internal returns (uint16 processedPeriod1, uint16 processedPeriod2) {
         uint16 currentPeriod = getCurrentPeriod();
-        uint16 previousPeriod = currentPeriod  - 1;
+        uint16 previousPeriod = currentPeriod - 1;
         StakerInfo storage info = stakerInfo[_staker];
 
         if (info.nextCommittedPeriod == 0 ||
             info.currentCommittedPeriod == 0 &&
             info.nextCommittedPeriod > previousPeriod ||
             info.currentCommittedPeriod > previousPeriod) {
-            return;
+            return (0, 0);
```

Comment:
This is the only `return` in this function. Can this function optionally not return the two periods?

Reply:
If no `return` statement - then just returns last value of variable (`processedPeriod1`, `processedPeriod2`). For example, assigning at line 1182